src/linesearches_compat.jl

```diff
@@ -0,0 +1,234 @@
+
+"""
+    LineSearchesStepsize <: Stepsize
+
+Wrapper for line searches available in the `LineSearches.jl` library.
+
+## Constructors
+
+    LineSearchesStepsize(
+        M::AbstractManifold,
+        ls,
+        is;
+        retraction_method::AbstractRetractionMethod=default_retraction_method(M),
+        vector_transport_method::AbstractVectorTransportMethod=default_vector_transport_method(M),
+    )
+    LineSearchesStepsize(
+        ls,
+        is;
+        retraction_method::AbstractRetractionMethod=default_retraction_method(M),
+        vector_transport_method::AbstractVectorTransportMethod=default_vector_transport_method(M),
+    )
+
+Wrap linesearch `ls` (for example `HagerZhang` or `MoreThuente`) and initial step selector
```

Comment:
Is there something we can link to? Is it `Optim.HagerZhang()`? Then maybe that also helps?

Reply:
Sure, I will add a link to docs.

Reply:
Done.